src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1734,11 +1742,23 @@ private String launchCommandIn(ArgumentListBuilder args, File workDir, EnvVars e
         String command = gitExe + " " + StringUtils.join(args.toCommandArray(), " ");
         try {
             args.prepend(gitExe);
-            listener.getLogger().println(" > " + command + (timeout != null ? TIMEOUT_LOG_PREFIX + timeout : ""));
+
+            if (timeout == null) {
+                Jenkins jenkins = Jenkins.getInstance();
+                if (jenkins != null) {
+                    GitTool.DescriptorImpl gitTool = (GitTool.DescriptorImpl) jenkins.getDescriptor(GitTool.class);
+                    if (gitTool != null && gitTool.getGitDefaultTimeout() != null && gitTool.getGitDefaultTimeout() > 0) {
+                        timeout = gitTool.getGitDefaultTimeout();
+                    } else {
+                        timeout = TIMEOUT;
+                    }
+                }
+            }
+            listener.getLogger().println(" > " + command + (TIMEOUT_LOG_PREFIX + timeout));
```

Comment:
Could you add one or more tests to assure that a new default timeout has been honored in the call to command line git?  [GitAPITestCase](https://github.com/jenkinsci/git-client-plugin/blob/master/src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestCase.java#L110) includes several tests which check that the timeout values are being correctly echoed in the command line git output.

I would also be curious if setting the global timeout will apply to command line git calls like "git clean".  There are one or more requests to set a longer timeout for "git clean" when using a large repository over a slow file system.